hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/security/acl/TestOzoneAuthorizerFactory.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.ozone.security.acl;
+
+import org.apache.hadoop.hdds.conf.OzoneConfiguration;
+import org.apache.hadoop.ozone.om.OzoneManager;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+import javax.annotation.Nonnull;
+
+import static org.apache.hadoop.ozone.OzoneConfigKeys.OZONE_ACL_AUTHORIZER_CLASS;
+import static org.apache.hadoop.ozone.OzoneConfigKeys.OZONE_ACL_ENABLED;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertInstanceOf;
+import static org.junit.jupiter.api.Assertions.assertNotSame;
+import static org.junit.jupiter.api.Assertions.assertSame;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class TestOzoneAuthorizerFactory {
+
+  @Test
+  void aclsDisabled() {
+    // GIVEN
+    OzoneManager om = disableAcls();
+
+    // WHEN
+    IAccessAuthorizer omAuth =
+        OzoneAuthorizerFactory.forOM(om);
+
+    // THEN
+    assertSame(OzoneAccessAuthorizer.get(), omAuth);
+
+    assertSameInstanceForSnapshot(om, omAuth);
+  }
+
+  @ParameterizedTest
+  @ValueSource(classes = {
+      OzoneNativeAuthorizer.class,
+      MockNativeAuthorizer.class,
+  })
+  void nativeAuthorizer(Class<? extends IAccessAuthorizer> clazz) {
+    // GIVEN
+    OzoneManager om = enableAcls(clazz);
+
+    // WHEN
+    IAccessAuthorizer omAuth =
+        OzoneAuthorizerFactory.forOM(om);
+
+    // THEN
+    assertInstanceOf(clazz, omAuth);
+
+    assertNewInstanceForSnapshot(om, omAuth);
+  }
+
+  @Test
+  void customNativeAuthorizer() {
```

Comment:
How does this test differ from the parameterized one above using the second parameter,       MockNativeAuthorizer.class,

Reply:
Well spotted, it doesn't.